src/main/kotlin/org/rust/cargo/runconfig/test/CargoTestEventsConverter.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.cargo.runconfig.test
+
+import com.google.gson.*
+import com.intellij.execution.testframework.TestConsoleProperties
+import com.intellij.execution.testframework.sm.ServiceMessageBuilder
+import com.intellij.execution.testframework.sm.runner.OutputToGeneralTestEventsConverter
+import com.intellij.openapi.util.Key
+import jetbrains.buildServer.messages.serviceMessages.*
+
+data class LibtestSuiteMessage(
+    val type: String,
+    val event: String
+) {
+    companion object {
+        fun fromJson(json: JsonObject): LibtestSuiteMessage? {
+            if (json.getAsJsonPrimitive("type")?.asString != "suite") {
+                return null
+            }
+
+            return Gson().fromJson(json, LibtestSuiteMessage::class.java)
+                ?: error("Failed to parse LibtestSuiteMessage from $json")
+        }
+    }
+}
+
+data class LibtestTestMessage(
+    val type: String,
+    val event: String,
+    val name: String
+) {
```

Comment:
Interesting
I see `stdout` param with stacktrace for failed events in terminal, but can't find it in messages that we receive in IDE. It would be great to get such info too to pass stacktrace for each failed test